src/borg/archive.py

```diff
@@ -1976,10 +1977,27 @@ def save(self, archive, target, comment=None, replace_original=True):
             return
         if comment is None:
             comment = archive.metadata.get('comment', '')
-        target.save(comment=comment, stats=target.stats, additional_metadata={
+        duration = timedelta(seconds=time.monotonic() - target.start_monotonic)
+
+        # Keep for the statistics if necessary
+        if self.stats:
+            _start = target.start
+
+        if self.timestamp is None:
+            end = archive.metadata.get('time_end') or archive.metadata.time
+            start = archive.metadata.time
+            logger.info(archive.metadata.time)
```

Comment:
guess you added this for debugging.

Reply:
Hum I forgot to remove the logger yes.
For what is left, actually no:
Since we need to keep the good metadata for the statistics, we want to have either the old metadata (L1987 and 1988). 

We need to keep `_start` (which is the date of creation of the new `Archive`) for the statistics: I guess we want to have the statistics of the current run, or are they supposed to be the one with the recreation date ?
If so indeed my code is wrong, and I can remove this `_start`.

Reply:
ok, guess i misunderstood it.

it is fine to have the recreate duration (start, end) for the --stats output as this refers to the current operation.

if no --timestamp is used, the created archive has the same start/end as it had before. so the duration (end-start) refers to the time needed for the original backup.

if --timestamp is used, the duration should follow the same semantics I guess or it will be inconsistent to the other case.
for the created archive, we want to have the given --timestamp in the metadata,